Bottom: d551dadf671871078fc6a8314f4015e8e5d6e78a
Top:    f1dcc179b934b653b4de6f28bbbed627b9fc6acd
Author: Sean Hefty <sean.hefty@intel.com>
Date:   2012-05-21 16:39:46 -0700

rstream: Check for connection error on async connect

Signed-off-by: Sean Hefty <sean.hefty@intel.com>


---

diff --git a/examples/rstream.c b/examples/rstream.c
index 104b318..c440f04 100644
--- a/examples/rstream.c
+++ b/examples/rstream.c
@@ -448,7 +448,8 @@ static int client_connect(void)
 {
 	struct addrinfo *res;
 	struct pollfd fds;
-	int ret, rs;
+	int ret, rs, err;
+	socklen_t len;
 
  	ret = getaddrinfo(dst_addr, port, NULL, &res);
 	if (ret) {
@@ -477,6 +478,17 @@ static int client_connect(void)
 		ret = do_poll(&fds);
 		if (ret)
 			goto err;
+
+		len = sizeof err;
+		ret = rs_getsockopt(rs, SOL_SOCKET, SO_ERROR, &err, &len);
+		if (ret)
+			goto err;
+		if (err) {
+			ret = -1;
+			errno = err;
+			perror("async rconnect");
+			goto err;
+		}
 	}
 
 free:
